pkg/backends/influxdb/influxql/unmarshal.go

```diff
@@ -99,29 +99,29 @@ func UnmarshalTimeseriesReader(reader io.Reader, trq *timeseries.TimeRangeQuery)
 			var sz int64
 			var wg sync.WaitGroup
 			errs := make([]error, len(wfd.Results[i].SeriesList[j].Values))
-			wg.Add(len(wfd.Results[i].SeriesList[j].Values))
 			for vi, v := range wfd.Results[i].SeriesList[j].Values {
-				go func(vals []any, idx int) {
+				wg.Go(func() {
+					var (
+						vals  = v
```

Comment:
@Elia-Renzoni I don't think the loop redeclarations like this are necessary. As of go 1.22, `for` loop variables [are properly assigned per-iteration](https://go.dev/blog/loopvar-preview), so they should be concurrency-safe when passed into goroutines.